### DISCUSSION TITLE
How to use 'b.run_compute()' to fit the multi-band data ?

### DISCUSSION THREAD

**Discussion by YmcGW** [phoebe-project/phoebe2#664] [Questions]
Hi!  I got this TypeError: buffer is too small for requested array, when I used 'b.run_compute()' to fit the multi-band data. I have abb three datasets  and set some parameters like this 
![data2](https://user-images.githubusercontent.com/118886353/203507225-7ae149e6-ecc2-4e17-8713-23b379003234.png)
So I would like to ask how to set the parameters for multi-band dataset .

**Comment by kecnry** [comment]
Your script looks reasonable, although if you're trying to get information out of the colors and your data are calibrated, then you'll likely want to make use of [dataset-coupled pblums](http://phoebe-project.org/docs/2.4/tutorials/pblum).

As for the buffer error message, can you please post the whole traceback so that we can see where it's occurring as well as which version of PHOEBE you have installed?  Does the error message go away if you use a single passband?  What if you use 10 points instead of 50 in `compute_phases`?

**Comment by YmcGW** [comment]
This is the whole traceback, and I've divided it into two graphs .
![e1](https://user-images.githubusercontent.com/118886353/203760815-a30ce316-c865-4307-abf4-f4bb254ba64a.png)
![e2](https://user-images.githubusercontent.com/118886353/203760828-1e049432-d711-4157-80fd-f141605f5c0a.png)
I also try to use10 point in compute_phases , but it didn't get any better .

**Reply by kecnry** [reply]
Ok, it looks like it is occurring when loading the passbands themselves.  If you do a single passband at a time (use just "Johnson:B" then try with just "Johnson:V", etc), can you isolate the problem to one of the passbands or does it occur for all of them?

**Reply by YmcGW** [reply]
I can run 'b.run_compute()' sucessfully with just  "Johnson:B"and"Johnson:V",without "Johnson:R". And I try to use a single passband "Johnson:R",but failed. So what shouid I do to add the data of the "Johnson:R" passband .

**Reply by aprsa** [reply]
Johnson:R should definitely work and I will try to reproduce your problem, but are you sure you have the data in Johnson:R? That passband is exceedingly uncommon these days. Cousins R is much more typical, and is found in the Bessell UBVRI filter-set.

**Comment by aprsa** [comment]
I just tried to reproduce the issue and couldn't; might it be possible that the Johnson:R passband file was modified or somehow got corrupted? I'd try to re-download it from the server and see if the issue persists.

**Reply by kecnry** [reply]
@YmcGW - the easiest way to do this is probably by calling [phoebe.uninstall_passband('Johnson:R')](http://phoebe-project.org/docs/latest/api/phoebe.atmospheres.passbands.uninstall_passband) (which should delete the passband file in your installation) and then rerunning your script (which should then notice you don't have that file and re-fetch it from the server).

**Reply by YmcGW** [reply]
Because of your help,I can run 'b.run_compute()' sucessfully with "Johnson:R". Thank you so much .
